### PR TITLE
Code unification

### PR DESCRIPTION
**Pull request by rafal-gorecki**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,27 @@
-# rosbot-slam
+# rosbot-autonomy
 
-A step-by-step guide for the ROSbot 2R/PRO to map an unknown environment and navigate autonomously within it.
+A step-by-step guide for the ROSbot 2R/PRO to map an unknown environment and navigate autonomously within it from Rviz.
```
Comment (delihus) [inline README.md:3]:
```suggestion
A step-by-step guide for the ROSbot 2R/PRO to map an unknown environment and navigate autonomously within it from RViz.
```

File: README.md [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 
-## Clonning the repo
+## Repository Setup
 
-This repository contains the Docker Compose setup for both PC and ROSbot. You can clone it to both PC and ROSbot, or use the `./sync_with_rosbot.sh` script to clone it to your PC and keep it synchronized with the robot
+This repository contains the Docker Compose setup for both PC and ROSbot 2R/PRO. You can clone it to both PC and ROSbot 2R/PRO, or use the `./sync_with_rosbot.sh` script to clone it to your PC and keep it synchronized with the robot
```
Comment (delihus) [inline README.md:7]:
```suggestion
This repository contains the Docker Compose setup for both PC and ROSbot 2, 2R and 2 PRO. You can clone it to both PC and ROSbot 2, 2R and 2 PRO, or use the `./sync_with_rosbot.sh` script to clone it to your PC and keep it synchronized with the robot
```

File: README.md [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+cd rosbot-autonomy
 export ROSBOT_ADDR=10.5.10.123 # Replace with your own ROSbot's IP or Husarnet hostname
 ./sync_with_rosbot.sh $ROSBOT_ADDR
 ```
 
-## Flashing the ROSbot Firmware
+## Flashing the ROSbot's Firmware
 
-Execute in the ROSbot's shell:
+To flash the Micro-ROS based firmware for STM32F4 microcontroller responisble for low-level functionalities of ROSbot 2R/PRO, execute in the ROSbot's shell:
```
Comment (delihus) [inline README.md:18]:
```suggestion
To flash the Micro-ROS based firmware for STM32F4 microcontroller responisble for low-level functionalities of ROSbot 2, 2R and 2 PRO, execute in the ROSbot's shell:
```

File: README.md [hunk 4]
```diff
@@ -68,38 +80,31 @@ LIDAR_BAUDRATE=256000
 
 The default options should be suitable.
 
-## Switching Between **SLAM** and **Localization Only** Mode
+## I. Running on a Physical Robot
+
+### ROSbot 2R/PRO
```
Comment (delihus) [inline README.md:85]:
```suggestion
### ROSbot 2, 2R and 2 PRO 
```

File: README.md [hunk 5]
```diff
@@ -68,38 +80,31 @@ LIDAR_BAUDRATE=256000
...
-```bash
-xhost +local:docker && \
-docker compose -f compose.pc.yaml up
-```
 
-### Localization & navigation
+#### Option 2: Localization Mode
 
-To allow the ROSbot to localize on a previously created map using AMCL, run:
+To allow the ROSbot 2R/PRO to localize on a previously created map using AMCL, run:
```
Comment (delihus) [inline README.md:103]:
```suggestion
To allow the ROSbot 2, 2R and 2 PRO to localize on a previously created map using AMCL, run:
```

File: compose.yaml [hunk 6]
```diff
@@ -7,26 +7,39 @@ x-net-config:
...
       - ${SERIAL_PORT:?err}
-    command: ros2 run micro_ros_agent micro_ros_agent serial -D $SERIAL_PORT serial -b 576000 # -v6
+    environment: #only two envs are valid for micro-xrce-agent
+      - FASTRTPS_DEFAULT_PROFILES_FILE            # ***
+      - XRCE_DOMAIN_ID_OVERRIDE=${ROS_DOMAIN_ID:-}  # ***
+    command: MicroXRCEAgent serial -D $SERIAL_PORT serial -b 576000 # -v6
 
   rplidar:
-    image: husarion/rplidar:humble-nightly
+    image: husarion/rplidar:humble
```
Comment (delihus) [inline compose.yaml:25]: Please add an exact tag of the `rplidar` docker image.

File: README.md [hunk 7]
```diff
@@ -68,38 +80,31 @@ LIDAR_BAUDRATE=256000
...
 
 ```bash
 SLAM_MODE=localization docker compose up -d
 ```
-To run the navigation stack, run on your **PC**:
-```bash
-docker compose -f compose.pc.yaml run navigation
-```
 
 > **Note:** You do not need to stop the containers to switch between modes.
```
Comment (delihus) [inline README.md:109]:
```suggestion
> [!NOTE]
> You do not need to stop the containers to switch between modes.
```

**Review by delihus — CHANGES_REQUESTED** [review]
Please also add the mecanum variable like here https://github.com/husarion/rosbot-docker/tree/ros2/demo

File: compose.yaml (hunk 6 above)
Comment (rafal-gorecki) [inline compose.yaml:25]:
```suggestion
    image: husarion/rplidar:humble-1.0.1-20230427
```

File: .env [hunk 8]
```diff
@@ -1,10 +1,27 @@
...
+# - False - regular
+MECANUM=True
+
+# =======================================
+# Software configurration
+# =======================================
+
+# Select SLAM mode:
+# - slam - with mapping
+# - localization - without mapping
```
Comment (rafal-gorecki) [inline .env:21]:
```suggestion
# - localization - without mapping (to load map place the `map.yaml` file in the `./maps` path)
```

**Review by delihus — APPROVED** [review]

File: compose.yaml [hunk 9]
```diff
@@ -7,26 +7,29 @@ x-net-config:
...
 
   rosbot:
-    image: husarion/rosbot:humble
+    image: husarion/rosbot:humble-0.6.1-20230712
     <<: *net-config
-    command: ros2 launch rosbot_bringup bringup.launch.py
+    command: ros2 launch rosbot_bringup bringup.launch.py mecanum:=${MECANUM:-True}
 
   microros:
-    image: husarion/micro-ros-agent:humble
```
Comment (delihus) [inline compose.yaml:15]: please use micro ros agent

**Review by delihus — CHANGES_REQUESTED** [review]

File: net.env [hunk 10]
```diff
@@ -14,4 +14,11 @@ RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
...
 # 4. Cyclone DDS + VPN
 # RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
+# FASTRTPS_DEFAULT_PROFILES_FILE=/husarnet-fastdds.xml
 # CYCLONEDDS_URI=file:///husarnet-cyclonedds.xml
+
+# =======================================
+# Setup ROS_DOMAIN_ID for all containers
+# =======================================
+
+ROS_DOMAIN_ID=123
```
Comment (delihus) [inline net.env:24]: Set default ROS_DOMAIN_ID to 0

File: compose.yaml [hunk 11]
```diff
@@ -7,26 +7,26 @@ x-net-config:
...
+    image: husarion/micro-ros-agent:humble-3.1.3-20231013-stable
     <<: *net-config
     devices:
       - ${SERIAL_PORT:?err}
     command: ros2 run micro_ros_agent micro_ros_agent serial -D $SERIAL_PORT serial -b 576000 # -v6
 
   rplidar:
-    image: husarion/rplidar:humble-nightly
+    image: husarion/rplidar:humble-1.0.1-20230427
```
Comment (delihus) [inline compose.yaml:22]: use stable version husarion/rplidar:humble-1.0.1-20230629-stable

File: compose.yaml [hunk 12]
```diff
@@ -7,26 +7,26 @@ x-net-config:
 services:
 
   rosbot:
-    image: husarion/rosbot:humble
+    image: husarion/rosbot:humble-0.6.1-20230712
```
Comment (delihus) [inline compose.yaml:10]: use stable version

**Review by delihus — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -1,134 +1,100 @@
...
 
-Execute in the ROSbot's shell:
+To flash the Micro-ROS based firmware for STM32F4 microcontroller responisble for low-level functionalities of ROSbot 2, 2R and 2 PRO, execute in the ROSbot's shell:
 
 ```bash
 docker stop rosbot microros || true && docker run \
 --rm -it --privileged \
-husarion/rosbot:humble \
-/flash-firmware.py /root/firmware.bin
+husarion/rosbot:humble-0.6.1-20230712 \
```
Comment (delihus) [inline README.md:23]: use stable

**Review by delihus — APPROVED** [review]

**Pull request merged**